apps/tx-builder/src/utils.ts

```diff
@@ -102,3 +90,47 @@ export const isValidAddress = (address: string | null) => {
   }
   return isAddress(address);
 };
+
+const NON_VALID_CONTRACT_METHODS = ['receive', 'fallback'];
+
+export function encodeToHexData(contractMethod: ContractMethod | undefined, contractFieldsValues: any) {
+  const contractMethodName = contractMethod?.name;
+  const contractFields = contractMethod?.inputs || [];
+
+  const isValidContractMethod = contractMethodName && !NON_VALID_CONTRACT_METHODS.includes(contractMethodName);
+
+  if (isValidContractMethod) {
+    try {
+      const parsedValues = contractFields.map((contractField: ContractInput, index) => {
+        const contractFieldName = contractField.name || index;
+        const cleanValue = contractFieldsValues[contractFieldName] || '';
+
+        return parseInputValue(contractField, cleanValue);
+      });
+      const abi = abiCoder as unknown; // a bug in the web3-eth-abi types
+      const hexEncondedData = (abi as AbiCoder).encodeFunctionCall(contractMethod as AbiItem, parsedValues);
```

Comment:
Is it safe to cast these?

Reply:
yep it seems that its a bug related with the `web3-eth-abi` lib.